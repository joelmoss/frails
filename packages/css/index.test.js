const serializer = require('jest-serializer-path')
const { createConfig } = require('@frails/core')

const css = require('.')
jest.mock('./require_resolve')

describe('css', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer(serializer)
  })

  test('within createConfig', () => {
    expect(createConfig(css)).toMatchSnapshot()
    expect(JSON.stringify(createConfig(css()))).toBe(JSON.stringify(createConfig(css)))
  })

  test('disabled', () => {
    expect(createConfig(css(false))).toMatchSnapshot()
  })
})
