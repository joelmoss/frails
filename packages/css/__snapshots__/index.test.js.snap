// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css disabled 1`] = `
Object {
  "entry": Object {},
  "mode": "development",
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
  ],
  "resolve": Object {
    "alias": Object {
      "lib": "<PROJECT_ROOT>/app/lib",
    },
  },
}
`;

exports[`css within createConfig 1`] = `
Object {
  "entry": Object {},
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css\\$/i,
        "use": Array [
          "mini-css-extract-plugin",
          Object {
            "loader": "css-loader",
            "options": Object {
              "modules": Object {
                "localIdentName": "[local]-[md5:hash:hex:6]",
              },
            },
          },
        ],
      },
    ],
  },
  "output": Object {
    "path": "<PROJECT_ROOT>/public/assets",
    "publicPath": "/assets/",
  },
  "plugins": Array [
    WebpackFixStyleOnlyEntriesPlugin {
      "apply": [Function],
      "options": Object {
        "extensions": Array [
          "less",
          "scss",
          "css",
          "sass",
          "styl",
        ],
        "ignore": undefined,
        "silent": false,
      },
    },
    Object {},
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "[name].css",
        "filename": "[name].css",
        "ignoreOrder": false,
        "moduleFilename": [Function],
      },
    },
  ],
  "resolve": Object {
    "alias": Object {
      "lib": "<PROJECT_ROOT>/app/lib",
    },
  },
}
`;

exports[`css without args 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css\\$/i,
        "use": Array [
          "mini-css-extract-plugin",
          Object {
            "loader": "css-loader",
            "options": Object {
              "modules": Object {
                "localIdentName": "[local]-[md5:hash:hex:6]",
              },
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "[name].css",
        "filename": "[name].css",
        "ignoreOrder": false,
        "moduleFilename": [Function],
      },
    },
  ],
}
`;
